### PR TITLE
Filter notifications by reports with/without decision and reportable type

### PR DESCRIPTION
**Pull request by ncounter**
notifications-reports-filters

### CODE REVIEW

File: src/api/app/views/webui/users/notifications/_notification_filter.html.haml [hunk 1]
```diff
@@ -56,6 +56,15 @@
                                                             label_icon: 'fas fa-people-group',
                                                             checked: selected_filter[:kind]&.include?('member_on_groups') }
 
+  .mt-4.mb-4
+    %h6.px-3.py-2
+      %b Reports
+    .px-3
+      = render partial: 'webui/shared/select', locals: { filter_item: 'report_with_decision',
```
Comment (krauselukas) [inline src/api/app/views/webui/users/notifications/_notification_filter.html.haml:63]: @ncounter I would say we should also check for ` policy(Report.new).notify?` to render this only to the users who are in beta and actually use it

Comment (ncounter) [reply src/api/app/views/webui/users/notifications/_notification_filter.html.haml]: Good point, I added that :+1:

**Comment by hennevogel** [comment]
Why are those separated 3 filters?

**Comment by rubhanazeem** [comment]
Everything functionally works well. Would be even better to have some specs for the new filters https://github.com/openSUSE/open-build-service/blob/master/src/api/spec/controllers/webui/users/notifications_controller_spec.rb

**Comment by ncounter** [comment]
> Why are those separated 3 filters?

I'm not sure I get 100% the questioning, but here it is an explanation:
- the `reports` checkbox is a `Type` filter and stays there.
- Everything else goes in a separated block because they are _details_ or _specifications_ of a certain `Type`, represented by the filter block title.
  - the `with` and `without` decision are checkboxes because of the possible choice: none or both = all, with decision only, without decision only
  - the reportable type is something that targets a different filter perspective, it's independent from the decision filter

One improvement could be to auto select the `Type=reports` if anything of the new filter is selected, because the filtering is anyway implicitly applied though.

I hope I got your objection right. What would it be your suggestion instead?

**Comment by hellcp-work** [comment]
> I hope I got your objection right. What would it be your suggestion instead?

I would put the checkboxes inside of a dropdown like it's done for requests

**Comment by ncounter** [comment]
> > I hope I got your objection right. What would it be your suggestion instead?
> 
> I would put the checkboxes inside of a dropdown like it's done for requests

Something I'd have proposed to be done the opposite :sweat_smile: (I mean, extracting the nested ones from the requests too).

Speaking of learnability and memorability (UX wise), filters are many now, and I (as a user) don't know/remember by hard what are the options. If I just see it, then I know I can filter by some property, otherwise I need to _guess_ and open the dropdown to find that.

Let's bring this to a different place as it is out of this PR scope, and to keep consistency I'll start with doing it like in the requests anyway.

**Comment by ncounter** [comment]
@hennevogel @hellcp-work I squeezed everything into one single dropdown, you can see it in the PR description `.gif`

**Comment by ncounter** [comment]
> Everything functionally works well. Would be even better to have some specs for the new filters https://github.com/openSUSE/open-build-service/blob/master/src/api/spec/controllers/webui/users/notifications_controller_spec.rb

@rubhanazeem I added a couple of specs in the last commit.

**Review by rubhanazeem — APPROVED** [review]

**Pull request merged**